src/code.cloudfoundry.org/gorouter/route/pool_test.go

```diff
@@ -232,134 +296,158 @@ var _ = Describe("EndpointPool", func() {
 			})
-			pool.Put(endpoint)
-			Expect(pool.LoadBalancingAlgorithm).To(Equal(expectedLBAlgo))
-			Eventually(logger).Should(gbytes.Say(`invalid-endpoint-load-balancing-algorithm-provided-keeping-pool-lb-algo`))
 		})
-	})
 
-	Context("Load balancing algorithm of a updated endpoint", func() {
-		It("is will overwrite the load balancing algorithm of the endpoint and pool", func() {
-			pool := route.NewPool(&route.PoolOpts{
-				Logger:                 logger.Logger,
-				LoadBalancingAlgorithm: config.LOAD_BALANCE_RR,
-			})
+		Context("Load balancing algorithm of a updated endpoint", func() {
+			It("will overwrite the load balancing algorithm of the endpoint and pool", func() {
+				pool := route.NewPool(&route.PoolOpts{
+					Logger:                 logger.Logger,
+					LoadBalancingAlgorithm: config.LOAD_BALANCE_RR,
+				})
+
+				endpointOpts := route.EndpointOpts{
+					Host:                   "host-1",
+					Port:                   1234,
+					RouteServiceUrl:        "url",
+					LoadBalancingAlgorithm: config.LOAD_BALANCE_LC,
+				}
+
+				initalEndpoint := route.NewEndpoint(&endpointOpts)
 
-			endpointOpts := route.EndpointOpts{
-				Host:                   "host-1",
-				Port:                   1234,
-				RouteServiceUrl:        "url",
-				LoadBalancingAlgorithm: config.LOAD_BALANCE_LC,
-			}
+				pool.Put(initalEndpoint)
+				Expect(pool.LoadBalancingAlgorithm).To(Equal(config.LOAD_BALANCE_LC))
 
-			initalEndpoint := route.NewEndpoint(&endpointOpts)
+				endpointOpts.LoadBalancingAlgorithm = config.LOAD_BALANCE_RR
+				updatedEndpoint := route.NewEndpoint(&endpointOpts)
+
+				pool.Put(updatedEndpoint)
+				Expect(pool.LoadBalancingAlgorithm).To(Equal(config.LOAD_BALANCE_RR))
+			})
+		})
 
-			pool.Put(initalEndpoint)
-			Expect(pool.LoadBalancingAlgorithm).To(Equal(config.LOAD_BALANCE_LC))
+		Context("Hash-Based load balancing algorithm", func() {
 
-			endpointOpts.LoadBalancingAlgorithm = config.LOAD_BALANCE_RR
-			updatedEndpoint := route.NewEndpoint(&endpointOpts)
+			It("correctly propagates the load balancing algorithm and HashRoutingProperties of the newly created endpoint to the other endpoints of the pool ", func() {
+				// TODO: route.HashBased will be implemented at a later date
+				//	poolWithLBAlgoHB := route.NewPool(&route.PoolOpts{
+				//		Logger:                 logger.Logger,
+				//		LoadBalancingAlgorithm: config.LOAD_BALANCE_HB,
+				//		//HashRoutingProperties:
+				//	})
+				//	iterator := poolWithLBAlgoHB.Endpoints(logger.Logger, "", false, "none", "az")
+				//	Expect(iterator).To(BeAssignableToTypeOf(&route.HashBased{}))
+				//	Eventually(logger).Should(gbytes.Say(`endpoint-iterator-with-round-robin-lb-algo`))
+			})
 
-			pool.Put(updatedEndpoint)
-			Expect(pool.LoadBalancingAlgorithm).To(Equal(config.LOAD_BALANCE_RR))
+			It("overwrites the load balancing algorithm and HashRoutingProperties of the endpoint and pool", func() {
+				//	TODO
```

Comment:
😄 I'm guessing this is intentional.

Reply:
okok, removed it... Will also be included in followup (maybe even w/o TODO 🤯 )